File: task_manager_back_end/src/main/java/edu/yasas/task_manager/dto/TaskDto.java
package edu.yasas.task_manager.dto;


import edu.yasas.task_manager.entity.UserEntity;

import lombok.Data;

import java.time.LocalDate;
import java.util.UUID;

@Data
public class TaskDto {

    private UUID id;

    private String title;

    private String description;

    private String status; //IN_PROGRESS,DONE,to do are the status

    private String priority; //HIGH,MEDIUM,LOW

    private LocalDate dueDate;

    private LocalDate createdAt;

    private LocalDate updatedAt;



}

File: task_manager_frontend/dto/taskDto.ts
import { TaskPriority, TaskStatus } from "@/types/task";
import { UserResponseDto } from "./user";

export interface TaskResponseDto{
    
    id: string;

    title: string;

    description: string;

    status: string; //IN_PROGRESS,DONE,to do are the status

    priority: string; //HIGH,MEDIUM,LOW

    dueDate: Date;

    createdAt: Date;

    updatedAt: Date;

    userResponseDto? : UserResponseDto;
}

export interface TaskRequestDto{
    
    title: string;

    description: string;

    status: string; //IN_PROGRESS,COMPLETED,TODO are the status

    priority: string; //HIGH,MEDIUM,LOW

    dueDate: Date;

    userId: string;
}

export interface TaskFormData {
  title: string;
  description: string;
  status: TaskStatus | string;
  priority: TaskPriority | string;
  dueDate: string;
}